packages/react-native/React/Base/RCTRootView.m

```diff
@@ -368,6 +368,9 @@ - (void)contentViewInvalidated
 - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
 {
   [super traitCollectionDidChange:previousTraitCollection];
+  if (RCTSharedApplication().applicationState == UIApplicationStateBackground) {
+    return;
+  }
 
```

Comment:
To confirm: where do we update the trait collection when the application foregrounds? If you change to dark mode while the app is backgrounded, do we pick it up on foreground?

Reply:
We'll get it on the foreground but we're in the foreground state before the user sees it so the change isn't visible to them

Reply:
The main issue I was trying to solve is when `_currentColorScheme` was set to nil initially, the default color scheme was `light` so on first render it would be `light` regardless of the users settings. So if the users device was in dark mode on initial render it would be reported as `light` until you toggled it and then it synced up

Reply:
Right, the changes to `_currentColorScheme` makes sense, I'm just a bit confused about how the RCTRootView changes are related here. Is it required as part of this PR?

> We'll get it on the foreground but we're in the foreground state

Where does this happen? Does `traitCollectionDidChange` when the application is foregrounded?